tests/test_metadata_utils.py

```diff
@@ -1641,6 +1679,102 @@ def test_add_metadata_and_chunk_examples_with_true_processor(self):
             "url: https://en.wikipedia.org/wiki/Apple ||| An <b>apple [[Malus domestica]]</b> is an edible fruit produced by an <a></a><b class:level1><b class:level2><i class:level3><a></a>apple</i> tree</b></b> (Malus domestica).<|endoftext|>",
         )
 
+    def test_add_metadata_and_chunk_examples_with_true_processor_and_metadata_special_tokens(self):
+        cfg = MetadataConfig()
+        cfg.metadata_list = ["url", "timestamp", "html", "entity"]
+        cfg.max_seq_len = 85
+        cfg.metadata_probability = 1
+        cfg.metadata_add_special_token_for_generation = True
+        cfg.metadata_special_token_for_generation_start_seq = " "
```

Comment:
adding this space allows metadata types to be always tokenised in the same way:
`" html entity"` -> `['Ġhtml', 'Ġentity']`
`" entity html"` -> `['Ġentity', 'Ġhtml']`